source/blog/2017-05-12-asynchronous-python-sdk.html.md

```diff
@@ -0,0 +1,127 @@
+
+## Asynchronous requests
+
+In order to the SDK to to work in an asynchronous fashion, we introduces two new features to our SDK, multiple connections and HTTP pipelining.
+This can bring the value when user want to fetch the inventory of the oVirt system. The time to fetch the inventory may be significantly decresed.
+You can see some comparison of the synchronous and asynchronous requests below.
+
+### Multiple connections
+
+Previously the SDK used only one opened connection which sequentially send the requests according to user program and always waited for the server response for corresponding request.
+In new version of the SDK the user can specify the number of connections the SDK should create to the server, and the specific requests created by user program use one of the connections.
+
+### HTTP pipelining
+
+As you can see at the image below, the HTTP requests are executed sequentially by default. The next request in order can be executed, when the previous request is received.
+With HTTP Pipelining client can send multiple requests without waiting for ther server response. Only idempotent HTTP methods can be pipelined.
+
+```
+With pipelining disabled            With pipelining enabled
+
+       |     |                            |      |
+       | \   |                            | \    |
+       |  \  |                            |  \   |
+       |   \ |                            | \ \  |
+       |    \|                            |  \ \ |
+       |    /|                            |   \ \|
+       |   / |                            |    \/| 
+       |  /  |                            |    /\|
+CLIENT | /   | SERVER              CLIENT |   / /| SERVER
+       | \   |                            |  / / |
+       |  \  |                            | / /  |
+       |   \ |                            |/ /   |
+       |    \|                            | /    |
+       |    /|                            |      |
+       |   / |                            
+       |  /  |                            
+       | /   |                            
+```
+
+## SDK implementation
+
+To not break the backward compatibility of the SDK, we have introduced a new boolean parameter called ```wait```, to the methods of SDK services.
+The default values is ```True```, so it's working in synchronous fashion. If the user send the ```wait=False```, the SDK will send a request specified by user program and will return the ```Future``` object,
+which is defined as follows:
+
+```python
+class Future(object):
+    """
+    Instances of this class are returned for operations that specify the
+    `wait=False` parameter.
+    """
+
+    def wait(self):
+        """
+        Waits till the result of the operation that created this future is
+        available.
+        """
+```
+
+So in order to wait for the response user need to call the ```wait``` method, which will return the resultion object.
```

Comment:
Rephrase: The user will need to call the wait method in order to wait for the response

resolution -> resulting  (I think)